### PR TITLE
add users registrations

### PR DESCRIPTION
**Pull request by kopchak**
https://trello.com/c/dRVLmkHf/10-user-registration

### CODE REVIEW

File: app/controllers/shareconomy/api/v1/auth/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+module Shareconomy::Api::V1::Auth
+  class RegistrationsController < DeviseTokenAuth::RegistrationsController
+    resource_description do
+      short 'Users manager'
+      api_versions 'v1'
+    end
+
+    api! 'Email registration. A verification email will be sent to the email address provided.'
+    param :email, String, required: true, desc: 'Email address'
+    param :first_name, String, required: true, desc: 'First name'
```
Comment (GeorgeZhukov) [inline app/controllers/shareconomy/api/v1/auth/registrations_controller.rb]: The profile model has no validation for this field, but here you point it as required.

File: app/controllers/shareconomy/api/api_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    rescue_from ActiveRecord::RecordNotFound do |exception|
+      response = Response.new
+      response.add_error_message(exception.message)
+      render json: response, status: :not_found
+    end
+
+    def create_entity(entity)
+      if entity.save
+        render json: Response.new(entity, scope: current_user), status: :created
```
Comment (kolyastepanets) [inline app/controllers/shareconomy/api/api_controller.rb]: may be current_shareconomy_user?

File: app/controllers/shareconomy/api/v1/auth/registrations_controller.rb (hunk 1 above)
Comment (kopchak) [inline app/controllers/shareconomy/api/v1/auth/registrations_controller.rb]: fixed

**Comment by kolyastepanets** [comment]
ok

**Comment by GeorgeZhukov** [comment]
ok

**Pull request merged**